src/main/java/calculator/StringAddCalculator.java

```diff
@@ -0,0 +1,51 @@
+package calculator;
+
+import java.util.Arrays;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class StringAddCalculator {
+    private static final int DEFAULT_RESULT = 0;
+    private static final String DEFAULT_DELIMITER_REGEX = ",|:";
+    private static final String CUSTOM_DELIMITER_REGEX = "//(.)\n(.*)";
+
+    public static int splitAndSum(final String text) {
+        if (nullOrEmpty(text)) {
+            return DEFAULT_RESULT;
+        }
+
+        final String[] tokens = tokens(text);
+
+        if (containsNegative(tokens)) {
+            throw new RuntimeException();
+        }
+
+        return sum(tokens);
+    }
+
+    private static boolean nullOrEmpty(final String text) {
+        return text == null || text.isEmpty();
+    }
+
+    private static String[] tokens(final String text) {
+        final Matcher m = Pattern.compile(CUSTOM_DELIMITER_REGEX).matcher(text);
```

Comment:
해당 메서드가 호출될때마다 패턴을 컴파일하도록 되어있네요!
패턴 컴파일은 시간이 많이 걸리는 프로세스이며 자원을 많이 사용하는것으로 알고 있습니다.
그렇기에 자바 doc에서도 한번 컴파일 후 재사용 할것을 권장하고 있습니다.
```
If a pattern is to be used multiple times, compiling it once and reusing 
it will be more efficient than invoking this method each time
```

불필요한 객체 생성을 피하도록 캐싱하여 개선하면 좋을 것 같아요 :)